### PR TITLE
Adding a prow job to run k8s unit tests using master go

### PR DESCRIPTION
**Pull request by Rajalakshmi-Girish**
This prow job is to trigger k8s unit tests whenever there is a change in upstream golang code. 

This job fetches the latest master golang for ppc64le from IBM storage  and sets up the master go before running the k8s Unit tests. uses make test to run UT.

https://prow.k8s-prow-c7f9fbfa2832409e2a3a6666e9dc52a3-0000.us-south.containers.appdomain.cloud/job-history/s3/prow-logs/logs/periodic-golang-master-build-ppc64le is responsible for maintaining the latest master go binary in IBM Storage

### CODE REVIEW

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ppc64le-cloud/test-infra/pull/91#" title="Author self-approved">Rajalakshmi-Girish</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **mkumatag** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @mkumatag` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ppc64le-cloud%2Ftest-infra).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/ppc64le-cloud/test-infra/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by Rajalakshmi-Girish** [comment]
\assign @mkumatag

File: config/jobs/ppc64le-cloud/builds/kubernetes-postsubmit.yaml
Comment (mkumatag) [inline config/jobs/ppc64le-cloud/builds/kubernetes-postsubmit.yaml:73]: Is this enough to generate the artifact? and will it dump the content to `ARTIFACT` folder?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: config/jobs/ppc64le-cloud/builds/kubernetes-postsubmit.yaml
Comment (Rajalakshmi-Girish) [reply config/jobs/ppc64le-cloud/builds/kubernetes-postsubmit.yaml:73]: Yes it is.
I see the junit xml file being created in artifacts folder where other existing jobs usually create. Please see the sample run that I pushed to gcs bucket:
https://console.cloud.google.com/storage/browser/ppc64le-kubernetes/logs/postsubmit-master-golang-kubernetes-unit-test-ppc64le/1331107525730242560/artifacts;tab=objects?pageState=(%22StorageObjectListTable%22:(%22f%22:%22%255B%255D%22))&authuser=0&project=ibm-kubernetes&prefix=&forceOnObjectsSortingFiltering=false

Also view the sample run in prow dashboard:
https://prow.k8s-prow-c7f9fbfa2832409e2a3a6666e9dc52a3-0000.us-south.containers.appdomain.cloud/view/gs/ppc64le-kubernetes/logs/postsubmit-master-golang-kubernetes-unit-test-ppc64le/1331107525730242560

Comment (Rajalakshmi-Girish) [reply config/jobs/ppc64le-cloud/builds/kubernetes-postsubmit.yaml:73]: hack/make-rules/test.sh from kubernetes has the below code when KUBE_JUNIT_REPORT_DIR is not passed:

```
KUBE_JUNIT_REPORT_DIR=${KUBE_JUNIT_REPORT_DIR:-}
# If KUBE_JUNIT_REPORT_DIR is unset, and ARTIFACTS is set, then have them match.
if [[ -z "${KUBE_JUNIT_REPORT_DIR:-}" && -n "${ARTIFACTS:-}" ]]; then
    export KUBE_JUNIT_REPORT_DIR="${ARTIFACTS}"
fi

Comment (mkumatag) [reply config/jobs/ppc64le-cloud/builds/kubernetes-postsubmit.yaml:73]: okay

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ppc64le-cloud/test-infra/pull/91#issuecomment-733642592" title="Approved">mkumatag</a>*, *<a href="https://github.com/ppc64le-cloud/test-infra/pull/91#" title="Author self-approved">Rajalakshmi-Girish</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ppc64le-cloud%2Ftest-infra).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ppc64le-cloud/test-infra/blob/master/OWNERS)~~ [mkumatag]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by ltccci** [comment]
@Rajalakshmi-Girish: Updated the `job-config` configmap in namespace `prow` at cluster `default` using the following files:
 - key `kubernetes-postsubmit.yaml` using file `config/jobs/ppc64le-cloud/builds/kubernetes-postsubmit.yaml`

<details>

In response to [this](https://github.com/ppc64le-cloud/test-infra/pull/91):

>This prow job is to trigger k8s unit tests whenever there is a change in upstream golang code. 
>
>This job fetches the latest master golang for ppc64le from IBM storage  and sets up the master go before running the k8s Unit tests. uses make test to run UT.
>
>https://prow.k8s-prow-c7f9fbfa2832409e2a3a6666e9dc52a3-0000.us-south.containers.appdomain.cloud/job-history/s3/prow-logs/logs/periodic-golang-master-build-ppc64le is responsible for maintaining the latest master go binary in IBM Storage


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Pull request merged**
